home-assignments/session1/solution1.py

```diff
@@ -0,0 +1,77 @@
+import json
+import yaml
+import argparse
+
+
+# TODO: fix encoding issue
+
+
+def import_data_from_file(file_name):
+    j_file = file_name
+    try:
+        with open(j_file, encoding="utf-8") as jfile:
+            output = json.load(jfile)
+
+            return output
+    except Exception as e:
+        print(e)
+        exit(1)
+
+
+def analyze_input_data(**output):
+    ppl_ages = output['ppl_ages']
+    buckets = output['buckets']
+    all_ages = []
+    for name, age in ppl_ages.items():
+        all_ages.append(age)
+
+    max_age = max(all_ages)
+    max_partition = max(buckets)
+    if max_age > max_partition:
+        buckets.append(max_age)
+
+    buckets.append(0)
+    buckets = sorted(buckets)
+    results = (ppl_ages, buckets)
+    return results
+
+
+def sort_output_data(data):
+    buckets_list = data[1]
+    pairs = []
+    for i in range(0, len(buckets_list) - 1):
+        pairs.append([buckets_list[i], buckets_list[i + 1]])
+
+    result_dict = {}
+    for pair in pairs:
```

Comment:
You can do 1 for loop instead of 2 loops. You can marge loops in line 42 + line 46, and right after line 43 do line 47..